### ISSUE TITLE
iCloud.com webpage does not open

### ISSUE THREAD

**Issue by NeilWaldhauer** [bitwiseworks/mozilla-os2#191]
When I go to iCoud.com to sync my iPhone, the page generates an error rather than appearing.

History: This worked for some time in Beta6, then it stopped. Then in Beta7 it work for some time, then it stopped. When I say worked, all the applications in iCoud work, including keeping web page, and iPhone or iPad synchronized.

The following is the Web console for the session
----------------------------------------
`The stylesheet https://edge.icloud.com/perf.css was not loaded because its MIME type, "text/plain", is not "text/css". www.icloud.com
"
                                   *********                           
                             *******       *******                     
                           ***                    ***                  
                         **                         **                 
                *****   **                           ***               
             ****   *****                              **              
           **                                           **             
          *                                              *             
         **              **                **            *             
      ****               **        **      **           ******         
    **                   **        **      **                 ****     
  ***                              **                            ***   
  *                                **                              **  
 *                               ****                               *  
 *                                                                  *  
**                                                                  ** 
*                            **         **                           * 
**                            ***********                            * 
 *                                ***                               ** 
  **                                                               **  
   ***                                                            **   
      ****                                                     ****    
         ****                                               ****       
            *************************************************          
 
 Happy to see you here! We are hiring the sharpest minds in the
 industry who want to build the most sophisticated and delightful
 applications on the web.
 
 Check out https://www.icloud.com/jobs/
 
" javascript-packed.js line 7 > eval:4:0
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create cloudkit.js:13:27383
Successfully compiled asm.js code (total compilation time 14ms; unable to cache asm.js in synchronous scripts; try loading asm.js via <script async> or createElement('script')) javascript.js
Successfully compiled asm.js code (total compilation time 11ms; unable to cache asm.js in synchronous scripts; try loading asm.js via <script async> or createElement('script')) javascript.js
Successfully compiled asm.js code (total compilation time 14ms; unable to cache asm.js in synchronous scripts; try loading asm.js via <script async> or createElement('script')) javascript.js
Successfully compiled asm.js code (total compilation time 4ms; not stored in cache (too small to benefit)) javascript.js
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead. javascript-packed.js:25:0
"CloudOS:  validate call failed with 0 status code" javascript-packed.js:111:461
"CloudOS:  SC.Benchmark "AUTH_DURATION" ended without a matching start.  No information was saved." javascript-packed.js:109:39
"CloudOS:  Session stat does not exist for cloudos" javascript-packed.js:111:461
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://icloud.cdn-apple.com/setup/ws/1/logout?clientBuildNumber=16GHotfix7&clientId=342DD895-5AAC-4BBE-925D-BB961598B06B&clientMasteringNumber=16GHotfix7&proxyDest=p01-setup&token=null. (Reason: CORS header 'Access-Control-Allow-Origin' missing). <unknown>
`

**Comment by SilvanScherrer** [comment]
is this still the case?

**Comment by dmik** [comment]
Well, given that this bug is just 14 days old I guess yes :) But I can confirm such a problem with 38.8.0 Beta 7. 38.2.1 Beta 6 shows exactly the same behaviour. However, in 31.8.0 Beta 5 it works well (despite the fact that it also complains about "text/plain" in the console — which proves it's not in vain here). A clean profile doesn't make any difference.

Anyway, we will see hoe esr45 behaves here and then decide what to do.

**Comment by NeilWaldhauer** [comment]
I can confirm that 31.8.0 Beta 5 works here. I can log in and run most apps, but Notes fails. I agree with waiting for esr45 and I will comment on how that works for me.

**Comment by NeilWaldhauer** [comment]
I discovered that I am starting Firefox with the Run!.exe program with only the !K option. I changed that to !LK, in order to run Firefox with LIBPATHSTRICT. Now iCloud works well, including Notes. This may be something for the readme, rather than any deep debugging.

**Comment by SilvanScherrer** [comment]
did you try Firefox 45 already?

**Comment by NeilWaldhauer** [comment]
On 05/22/17 05:32 am, Silvan Scherrer wrote:
>
> did you try Firefox 45 already?
>
>
I tried logging in and logging out, and iCloud seems to work. Notes 
functions and Calendar actually seems to run better. The issue I 
originally reported seems to be fixed.

iCloud is a moving target. Apple will replace login with two factor 
authentication on June 15. This used to fail on firefox 38, and I have 
not yet tried it on Firefox 45.

Thanks for this release. I have a bug to report regarding a regression 
displaying some graphics once I get organized. I hope I didn't cause the 
bug myself. In my rush to see Firefox 45, I enabled netlabs-exp.

Thanks also for supporting Warpstock EU. I missed your presentation, but 
as a VOICE board member I'd like to see Warpstock EU prosper.

best regards,
Neil

-- 
Neil Waldhauer  neil@blondeguy.com  www.blondeguy.com
Expert consulting for OS/2 Warp, ArcaOS and eComStation
